### ISSUE TITLE
Missing vfat support in kernel

### ISSUE THREAD

**Issue by rgaudin** [docker/for-mac#7102]
### Description

In version 4.26.0, the bundled linuxkit kernel lacks support for the `vfat` filesystem. It's not available as a module neither.
This change is not mentioned in the Changelog.
This is breaking a lot of my docker usages. I had to downgrade.

### Reproduce

```
❯ docker run --rm -it busybox sh -c "uname -a && cat /proc/filesystems |grep vfat"
Linux 13f735ffb85c 6.5.11-linuxkit #1 SMP PREEMPT_DYNAMIC Mon Dec  4 10:03:25 UTC 2023 x86_64 GNU/Linux
```

### Expected behavior

Output should show `vfat`, indicating support for this filesystem in the kernel

### docker version

```bash
Client:
 Cloud integration: v1.0.35+desktop.5
 Version:           24.0.7
 API version:       1.43
 Go version:        go1.20.10
 Git commit:        afdd53b
 Built:             Thu Oct 26 09:04:20 2023
 OS/Arch:           darwin/amd64
 Context:           desktop-linux

Server: Docker Desktop 4.26.0 (130397)
 Engine:
  Version:          24.0.7
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.10
  Git commit:       311b9ff
  Built:            Thu Oct 26 09:08:02 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.25
  GitCommit:        d8f198a4ed8892c764191ef7b3b06d8a2eeb5c7f
 runc:
  Version:          1.1.10
  GitCommit:        v1.1.10-0-g18a0cb0
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client:
 Version:    24.0.7
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.12.0-desktop.2
    Path:     /Users/reg/.docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.23.3-desktop.2
    Path:     /Users/reg/.docker/cli-plugins/docker-compose
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     /Users/reg/.docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.21
    Path:     /Users/reg/.docker/cli-plugins/docker-extension
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  0.1
    Path:     /Users/reg/.docker/cli-plugins/docker-feedback
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v0.1.0-beta.10
    Path:     /Users/reg/.docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /Users/reg/.docker/cli-plugins/docker-sbom
  scan: Docker Scan (Docker Inc.)
    Version:  v0.26.0
    Path:     /Users/reg/.docker/cli-plugins/docker-scan
  scout: Docker Scout (Docker Inc.)
    Version:  v1.2.0
    Path:     /Users/reg/.docker/cli-plugins/docker-scout

Server:
 Containers: 18
  Running: 0
  Paused: 0
  Stopped: 18
 Images: 53
 Server Version: 24.0.7
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: d8f198a4ed8892c764191ef7b3b06d8a2eeb5c7f
 runc version: v1.1.10-0-g18a0cb0
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 6.5.11-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 16
 Total Memory: 5.793GiB
 Name: docker-desktop
 ID: 144e8452-0a0a-4357-8452-cf894f541960
 Docker Root Dir: /var/lib/docker
 Debug Mode: true
  File Descriptors: 39
  Goroutines: 58
  System Time: 2023-12-07T09:31:49.863163396Z
  EventsListeners: 9
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false
```

### Diagnostics ID

98DAE040-42FB-4E97-89FD-44AA7C76B8F0/20231207093204

### Additional Info

It works as expected with version `4.25.2 (129061)`

```
❯ docker run --rm -it busybox sh -c "uname -a && cat /proc/filesystems |grep vfat"
Linux 98fe4329af86 6.4.16-linuxkit #1 SMP PREEMPT_DYNAMIC Thu Nov 16 10:55:59 UTC 2023 x86_64 GNU/Linux
	vfat
```

**Comment by rpardini** [comment]
Confirmed. 4.26.0 has no vfat support, breaking my use case which is building Armbian UEFI images. UEFI requires a FAT32 /boot/efi filesystem.

**Comment by rpardini** [comment]
I reverted to 4.25.2 (via https://docs.docker.com/desktop/release-notes/#4252 download links), after completely removing 4.26.0 and a factory reset. vfat works.

**Comment by dgageot** [comment]
Sorry all for this issue. This will be fixed in 4.27.0. In the meantime, yes, you can use 4.25.2. Sorry again.

**Comment by dgageot** [comment]
@rgaudin Could you test Docker Desktop 4.27 and close this issue if it's fixed?

**Comment by rgaudin** [comment]
Works fine! Thank you.

**Comment by mijenix** [comment]
I can confirm too, mounting vfat works again.

**Comment by rpardini** [comment]
> This will be fixed in 4.27.0.

Confirmed fixed, thanks. 4.27.2 (137060) also still works.
